ApplicationManager.js

```diff
@@ -0,0 +1,159 @@
+    this.algo.optimize();
+  }
+
+  plotFunctionAndAxis() {
+    this.plot.clearAll();
+
+    if (this.plot.getType() === plotTypes.plot_1D) {
+      this.plot.draw(this.algoParams[paramNames.objectiveFunction], 4);
+    }
+
+    if (this.plot.getType() === plotTypes.contour_plot) {
+      const thresholds = paramsConfig[paramNames.threshold].values[this.plot.getType()].get_init_value();
+      const interpolation = paramsConfig[paramNames.interpolation].values[this.plot.getType()]
+        .get_init_value(null, this.objectiveFunction.name).value;
+      this.plot.draw(this.algoParams[paramNames.objectiveFunction], 4, thresholds, d3[interpolation])
+    }
+
+    this.plot.addAxis();
+  }
+
+  plotAlgoResults() {
+    this.plot.clearLines();
+
+    if (this.plot.getType() === plotTypes.plot_1D) {
+      this.plot.addLine(this.algo.getPath().map(index => [index[0], this.algoParams[paramNames.objectiveFunction](index)]));
+    }
+
+    if (this.plot.getType() === plotTypes.contour_plot) {
+      this.plot.addLine(this.algo.getPath());
+    }
+  }
+
+  setParam(paramName, value) {
+    if (!Object.values(paramNames).includes(paramName)) {
+      throw new Error("Invalid parameter name");
+    }
+
+    if (paramName === paramNames.algorithmName) {
+      this.resetParams(value.name, this.objectiveFunction.name, this.objectiveFunction.value);
+      this.resetParametersButtons();
+
+      this.plotAlgoResults();
+
+      return;
+    }
+
+    if (paramName === paramNames.objectiveFunction) {
+      this.algoParams[paramNames.objectiveFunction] = value;
+      this.objectiveFunction = value;
+      this.resetParams(this.algo.getName());
+
+      const xDomain = paramsConfig[paramNames.xDomain].values[this.plot.getType()]
+        .get_init_value(null, this.objectiveFunction.name);
+      const yDomain = paramsConfig[paramNames.yDomain].values[this.plot.getType()]
+        .get_init_value(null, this.objectiveFunction.name);
+      this.plot.setXDomain(xDomain);
+      this.plot.setYDomain(yDomain);
+
+      this.plotFunctionAndAxis();
+      this.plotAlgoResults();
+      this.resetParametersButtons();
+      return;
+    }
+
+    if (paramsConfig[paramName].input_type === inputTypes.dropdown) {
+      this.algoParams[paramName] = value.value;
+    } else {
+      this.algoParams[paramName] = value;
+    }
+    this.resetAlgo(this.algo.getName());
+    this.plotAlgoResults();
+  }
+
+  resetParametersButtons() {
+    d3.select("#algoParamsButtonSpace").selectAll("*").remove();
+    algorithmsConfig[this.algo.getName()].parameters.forEach(param => {
+      const { input_type: inputType } = paramsConfig[param];
+      const { domain, get_init_value: getInitValue } = paramsConfig[param].values[this.plot.getType()];
+      let initValue;
+      if (param === paramNames.objectiveFunction) {
+        initValue = this.objectiveFunction;
+      } else if (param === paramNames.algorithmName) {
```

Comment:
Is this condition useful? I understand that the algorithmName is not part of the algorithmsConfig parameters